### PR TITLE
Prevent mero fail when fails a single cluster

### PR DESCRIPTION
**Pull request by ElFantasma**
Now when a single cluster fails in Mero configuration it won't crash. Instead, it will keep working and be able to restore the cluster once it starts working again.

Was tested in a single instance in production, blocking connections to elasticache using `iptables` command. Mero raised an exception but kept working for other clusters, and when the connectivity was restored it recovered.

### CODE REVIEW

File: src/mero_conf.erl [hunk 1]
```diff
@@ -267,24 +287,26 @@ process_value({servers, {mfa, {Module, Function, Args}}}) ->
...
-get_elasticache_cluster_config(Host, Port) ->
+get_elasticache_cluster_config(Host, Port, Retries) ->
     get_elasticache_cluster_config(Host,
                                    Port,
                                    0,
+                                   Retries,
                                    mero_elasticache:get_cluster_config(Host, Port)).
 
-get_elasticache_cluster_config(_Host, _Port, _Retries, {ok, Entries}) ->
+get_elasticache_cluster_config(_Host, _Port, _Retry, _Retries, {ok, Entries}) ->
```
Comment (facundoolano) [inline src/mero_conf.erl:302]: this could use some better naming to distinguish the current retries from the max allowed retries

**Review by facundoolano — APPROVED** [review]

File: src/mero_conf_monitor.erl [hunk 2]
```diff
@@ -142,16 +142,25 @@ update_clusters([ClusterDef | OldClusterDefs],
     update_clusters(OldClusterDefs, NewClusterDefs);
 update_clusters([{ClusterName, OldAttrs} | OldClusterDefs],
```
Comment (polvorin) [inline src/mero_conf_monitor.erl:143]: Can we drop the assumption about the two lists here?.   So we can have clusters _added_  or _removed_  at runtime, just changing the app environment (config). 

I guess we can use set differences,   a cluster that is on the old set but not on the new:  terminate the cluster.   One  in the new but not on the old:  start it.   Things that match exactly:  do nothing   (case for changed clusters could be just let it be terminated and then started,  or had special condition to call restart_child/1 explicitly).

Handling failing cluster configs would be _not_ returning them so they are like not listed on the app environment.

Comment (ElFantasma) [reply src/mero_conf_monitor.erl:143]: Ok, working on that.

**Comment by ElFantasma** [comment]
Addressed @polvorin comment on dropping the assumption on having two lists of the same size. This won't allow yet to change mero configuration on runtime, but it is a step closer to that behavior.

This new version was again tested in production in a single instance, blocking memcache ips with iptables and unblocking to see it recover.

**Review by polvorin — APPROVED** [review]

**Pull request merged**